### ISSUE TITLE
Abort message but the computation goes on

### ISSUE THREAD

**Issue by lopippo** [toppic-suite/toppic-suite#6]
Greetings,

at step 4.5 Mass spectral identification using TopPIC of the tutorial, after having started the computation, I get this message in the terminal (I have started the _gui version of the program):

[W::fai_get_val] Reference DECOY_sp|Q8ZLB8|BCSC_SALTY not found in FASTA file, returning empty sequence
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string: construction from null is not valid
Aborted

That message does not prevent the _gui-based computation of attain its objective:

TopPIC 1.5.3
********************** Parameters **********************
Protein database file:                      	/home/rusconi/labo/research/top-down-toppic/tutorial-data/uniprot-st.fasta
Spectrum file:                              	/home/rusconi/labo/research/top-down-toppic/tutorial-data/st_1_ms2.msalign
Number of combined spectra:                 	1
Fragmentation method:                       	FILE
Search type:                                	TARGET+DECOY
Fixed PTMs BEGIN                            
Carbamidomethylation                        	57.021464	C
Fixed PTMs END                              
Use TopFD feature file:                     	True
Maximum number of unexpected modifications: 	1
Error tolerance for matching masses:        	15 ppm
Error tolerance for identifying PrSM clusters: 	1.2 Da
Spectrum-level cutoff type:                 	FDR
Spectrum-level cutoff value:                	0.01
Proteoform-level cutoff type:               	FDR
Proteoform-level cutoff value:              	0.01
Allowed N-terminal forms:                   	NONE,NME,NME_ACETYLATION,M_ACETYLATION
Maximum mass shift of modifications:        	500 Da
Minimum mass shift of modifications:        	-500 Da
Thread number:                              	1
E-value computation:                        	Generating function
Executable file directory:                  	/usr/bin
Start time:                                 	Sun Oct 30 21:20:31 2022
Version:                                    	1.5.3
********************** Parameters **********************
Non PTM filtering - started.
Non PTM filtering - processing 100%.       
Non PTM filtering - combining blocks started.
Non PTM filtering - combining blocks finished.
Non PTM filtering - finished.
Non PTM search - started.
Non PTM search - processing 1381 of 1381 spectra.
Non PTM search - finished.
One PTM filtering - started.
One PTM filtering - processing 100%.        
One PTM filtering - combining blocks started.
One PTM filtering - combining blocks finished.
One PTM filtering - finished.
One PTM search - started.
One PTM search - processing 1381 of 1381 spectra.
One PTM search - finished.
Merging PrSMs - started.
Merging PrSMs - finished.
E-value computation - started.
E-value computation - processing 1381 of 1381 spectra.
E-value computation - finished.
Top PrSM selecting - started
Top PrSM selecting - finished.
Finding PrSM clusters - started.
Finding PrSM clusters - processing 100%.   
Finding PrSM clusters - finished.
FDR computation - started. 
FDR computation - finished.
PrSM filtering by FDR - started.
PrSM filtering by FDR - finished.
Outputting PrSM table - started.
Outputting PrSM table - finished.
Generating PrSM xml files - started.
Generating xml files - processing 136 PrSMs.
Generating xml files - processing 46 prsms.
Generating xml files - preprocessing 56 proteoforms.
Generating xml files - processing 56 proteoforms.
Generating xml files - preprocessing 29 proteins.
Generating xml files - processing 29 proteins.
Generating PrSM xml files - finished.
Converting PrSM xml files to json files - started.
Converting xml files to html files - processing 223 of 223 files.
Converting PrSM xml files to json files - finished.
PrSM filtering by FORMFDR - started.
PrSM filtering by FORMFDR - finished.
Selecting top PrSMs for proteoforms - started.
Selecting top PrSMs for proteoforms - finished.
Outputting proteoform table - started.
Outputting proteoform table - finished.
Generating proteoform xml files - started.
Generating xml files - processing 131 PrSMs.
Generating xml files - processing 44 prsms.
Generating xml files - preprocessing 51 proteoforms.
Generating xml files - processing 51 proteoforms.
Generating xml files - preprocessing 28 proteins.
Generating xml files - processing 28 proteins.
Generating proteoform xml files - finished.
Converting proteoform xml files to html files - started.
Converting xml files to html files - processing 212 of 212 files.
Converting proteoform xml files to html files - finished.
********************** Parameters **********************
Protein database file:                      	/home/rusconi/labo/research/top-down-toppic/tutorial-data/uniprot-st.fasta
Spectrum file:                              	/home/rusconi/labo/research/top-down-toppic/tutorial-data/st_2_ms2.msalign
Number of combined spectra:                 	1
Fragmentation method:                       	FILE
Search type:                                	TARGET+DECOY
Fixed PTMs BEGIN                            
Carbamidomethylation                        	57	C
Fixed PTMs END                              
Use TopFD feature file:                     	True
Maximum number of unexpected modifications: 	1
Error tolerance for matching masses:        	15 ppm
Error tolerance for identifying PrSM clusters: 	1.2 Da
Spectrum-level cutoff type:                 	FDR
Spectrum-level cutoff value:                	0.01
Proteoform-level cutoff type:               	FDR
Proteoform-level cutoff value:              	0.01
Allowed N-terminal forms:                   	NONE,NME,NME_ACETYLATION,M_ACETYLATION
Maximum mass shift of modifications:        	500 Da
Minimum mass shift of modifications:        	-500 Da
Thread number:                              	1
E-value computation:                        	Generating function
Executable file directory:                  	/usr/bin
Start time:                                 	Sun Oct 30 21:21:45 2022
End time:                                   	Sun Oct 30 21:21:40 2022
Version:                                    	1.5.3
********************** Parameters **********************
Non PTM filtering - started.
Non PTM filtering - processing 100%.       
Non PTM filtering - combining blocks started.
Non PTM filtering - combining blocks finished.
Non PTM filtering - finished.
Non PTM search - started.
Non PTM search - processing 1532 of 1532 spectra.
Non PTM search - finished.
One PTM filtering - started.
One PTM filtering - processing 100%.        
One PTM filtering - combining blocks started.
One PTM filtering - combining blocks finished.
One PTM filtering - finished.
One PTM search - started.
One PTM search - processing 1532 of 1532 spectra.
One PTM search - finished.
Merging PrSMs - started.
Merging PrSMs - finished.
E-value computation - started.
E-value computation - processing 1532 of 1532 spectra.
E-value computation - finished.
Top PrSM selecting - started
Top PrSM selecting - finished.
Finding PrSM clusters - started.
Finding PrSM clusters - processing 100%.   
Finding PrSM clusters - finished.
FDR computation - started. 
FDR computation - finished.
PrSM filtering by FDR - started.
PrSM filtering by FDR - finished.
Outputting PrSM table - started.
Outputting PrSM table - finished.
Generating PrSM xml files - started.
Generating xml files - processing 177 PrSMs.
Generating xml files - processing 59 prsms.
Generating xml files - preprocessing 85 proteoforms.
Generating xml files - processing 85 proteoforms.
Generating xml files - preprocessing 48 proteins.
Generating xml files - processing 48 proteins.
Generating PrSM xml files - finished.
Converting PrSM xml files to json files - started.
Converting xml files to html files - processing 312 of 312 files.
Converting PrSM xml files to json files - finished.
PrSM filtering by FORMFDR - started.
PrSM filtering by FORMFDR - finished.
Selecting top PrSMs for proteoforms - started.
Selecting top PrSMs for proteoforms - finished.
Outputting proteoform table - started.
Outputting proteoform table - finished.
Generating proteoform xml files - started.
Generating xml files - processing 177 PrSMs.
Generating xml files - processing 59 prsms.
Generating xml files - preprocessing 85 proteoforms.
Generating xml files - processing 85 proteoforms.
Generating xml files - preprocessing 48 proteins.
Generating xml files - processing 48 proteins.
Generating proteoform xml files - finished.
Converting proteoform xml files to html files - started.
Converting xml files to html files - processing 312 of 312 files.
Converting proteoform xml files to html files - finished.
Merging files started.
Merging msalign files started.
Merging msalign files finished.
Merging json files started.
Copying file: "/home/rusconi/labo/research/top-down-toppic/tutorial-data/st_1_html/topfd/ms1_json/spectrum1417.js"
Copying file: "/home/rusconi/labo/research/top-down-toppic/tutorial-data/st_2_html/topfd/ms1_json/spectrum1536.js"
Copying file: "/home/rusconi/labo/research/top-down-toppic/tutorial-data/st_1_html/topfd/ms2_json/spectrum1380.js"
Copying file: "/home/rusconi/labo/research/top-down-toppic/tutorial-data/st_2_html/topfd/ms2_json/spectrum1531.js"
Merging json files finished.
Merging feature files started.
Merging feature files finished.
Merging identification files started.
Merging identification files finished.
Merging files - finished.
Finding PrSM clusters - started.
Finding PrSM clusters - processing 100%.   
Finding PrSM clusters - finished.
FDR computation - started. 
FDR computation - finished.
PrSM filtering by FDR - started.
PrSM filtering by FDR - finished.
Outputting PrSM table - started.
Outputting PrSM table - finished.
Generating PrSM xml files - started.
Generating xml files - processing 329 PrSMs.
Generating xml files - processing 110 prsms.
Generating xml files - preprocessing 130 proteoforms.
Generating xml files - processing 130 proteoforms.
Generating xml files - preprocessing 53 proteins.
Generating xml files - processing 53 proteins.
Generating PrSM xml files - finished.
Converting PrSM xml files to json files - started.
Converting xml files to html files - processing 514 of 514 files.
Converting PrSM xml files to json files - finished.
PrSM filtering by FORMFDR - started.
PrSM filtering by FORMFDR - finished.
Selecting top PrSMs for proteoforms - started.
Selecting top PrSMs for proteoforms - finished.
Outputting proteoform table - started.
Outputting proteoform table - finished.
Generating proteoform xml files - started.
Generating xml files - processing 308 PrSMs.
Generating xml files - processing 103 prsms.
Generating xml files - preprocessing 109 proteoforms.
Generating xml files - processing 109 proteoforms.
Generating xml files - preprocessing 48 proteins.
Generating xml files - processing 48 proteins.
Generating proteoform xml files - finished.
Converting proteoform xml files to html files - started.
Converting xml files to html files - processing 467 of 467 files.
Converting proteoform xml files to html files - finished.
Deleting temporary files - started.
Deleting temporary files - finished.
TopPIC finished.

I'll go on with the tutorial. One remark however: maybe it would be useful to associate the error messages with the filename and the source code line where the error occurred.

Sincerely,
Filippo

**Comment by liuxiaowen** [comment]
We will fix the issue.

**Comment by lopippo** [comment]
> We will fix the issue.

Great! Thank you!
Filippo

**Comment by liuxiaowen** [comment]
We tested toppic_gui 1.5.3 but could not reproduce the error. Can you also post the parameters you used in topindex_gui?

**Comment by lopippo** [comment]
I have posted them in the previous issue post as I was following bit by bit the section "4.5 Mass spectral identification using TopPIC" of the tutorial at https://www.toppic.org/software/toppic/tutorial.html.

However, the point is I cannot reproduce that message to the console today... I'll explore that a bit further. Closing this issue for the time being, then. Sorry for the noise.

Sincerely,
Filippo
